### ISSUE TITLE
Running linter with SL4

### ISSUE THREAD

**Issue by JeanMertz** [alecthomas/SublimeLinter-contrib-gometalinter#13]
Since upgrading to SL4, this package shows an error in the console:

```
  File "/Users/jean/Library/Application Support/Sublime Text 3/Packages/SublimeLinter-contrib-gometalinter/linter.py", line 39, in __init__
    gopath = self.get_view_settings().get('gopath')
  File "lint.linter in /Users/jean/Library/Application Support/Sublime Text 3/Installed Packages/SublimeLinter.sublime-package", line 383, in get_view_settings
RuntimeError: CRITICAL: gometalinter: Calling 'get_view_settings' outside of lint context
```

**Comment by JeanMertz** [comment]
When I temporarily "fix" the issue by using this patch:

```diff
diff --git a/linter.py b/linter.py
index 45f603c..068e6bb 100644
--- a/linter.py
+++ b/linter.py
@@ -36,14 +36,14 @@ class Gometalinter(Linter):
         if not self.env:
             self.env = os.environ.copy()
 
-        gopath = self.get_view_settings().get('gopath')
+        gopath = None # self.get_view_settings().get('gopath')
         if gopath:
             self.env['GOPATH'] = gopath
             print('sublimelinter: (custom) GOPATH={}'.format(self.env['GOPATH']))
         else:
             print('sublimelinter: (system) GOPATH={}'.format(os.environ.get('GOPATH', '')))
 
-        goroot = self.get_view_settings().get('goroot')
+        goroot = None # self.get_view_settings().get('goroot')
         if goroot:
             self.env['GOROOT'] = goroot
             print('sublimelinter: (custom) GOROOT={}'.format(self.env['GOROOT']))
```

It then errors related to creating a temporary directory:

```
Traceback (most recent call last):
  File "lint.backend in /Users/jean/Library/Application Support/Sublime Text 3/Installed Packages/SublimeLinter.sublime-package", line 97, in execute_lint_task
  File "lint.linter in /Users/jean/Library/Application Support/Sublime Text 3/Installed Packages/SublimeLinter.sublime-package", line 797, in lint
  File "/Users/jean/Library/Application Support/Sublime Text 3/Packages/SublimeLinter-contrib-gometalinter/linter.py", line 55, in run
    return self._live_lint(cmd, code)
  File "/Users/jean/Library/Application Support/Sublime Text 3/Packages/SublimeLinter-contrib-gometalinter/linter.py", line 67, in _live_lint
    return tmpdir(self, dir, cmd, files, self.filename, code)
  File "/Users/jean/Library/Application Support/Sublime Text 3/Packages/SublimeLinter-contrib-gometalinter/linter.py", line 97, in tmpdir
    os.link(f, target)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/jean/code/go/src/github.com/blendle/go-streamprocessor/stream/.gometalinterw7pg8s/producer.go'
```

**Comment by alecthomas** [comment]
I had the same problems. I've pushed a fix in version 4.0.0, try the latest.
